doc/scheduled-configurations/scheduled-configurations-hld.md

```diff
@@ -0,0 +1,583 @@
+        - [Example](#example)
+      - [SCHEDULED\_CONFIGURATIONS Table](#scheduled_configurations-table)
+        - [Example](#example-1)
+    - [YANG Model Enhancements](#yang-model-enhancements)
+      - [TIME\_RANGE Yang Model](#time_range-yang-model)
+      - [SCHEDULED\_CONFIGURATIONS Yang Model](#scheduled_configurations-yang-model)
+    - [CLI Commands](#cli-commands)
+      - [Config Scheduled Configuration](#config-scheduled-configuration)
+      - [Config Time Range](#config-time-range)
+      - [Show Time Range](#show-time-range)
+      - [Show Scheduled Configurations](#show-scheduled-configurations)
+  - [SAI API](#sai-api)
+  - [Warmboot and Fastboot Design Impact](#warmboot-and-fastboot-design-impact)
+  - [Restrictions/Limitations](#restrictionslimitations)
+  - [Testing Requirements/Design](#testing-requirementsdesign)
+    - [Unit Test cases](#unit-test-cases)
+    - [System Test cases](#system-test-cases)
+  - [Open/Action items](#openaction-items)
+  - [References](#references)
+
+## Revision
+
+Revision 1.3
+
+## Scope
+
+The scope of this document is to outline the High-Level Design (HLD) for the *Scheduled Configurations* feature in SONiC. This feature aims to set SONiC configurations based on a schedule defined by the user. It includes the ability to schedule configurations for absolute-time events as well as recurring time intervals.
+
+## Definitions/Abbreviations
+
+| **Term**    | **Definition**                                                                       |
+| ----------- | ------------------------------------------------------------------------------------ |
+| *CONFIG_DB* | Configuration Database used by SONiC to store the static configuration of the system |
+| *STATE_DB*  | State Database in SONiC to store the runtime state of the system                     |
+| *SWSS*      | SONiC SWitch State service                                                           |
+| *cron*      | time-based job scheduler linux application                                           |
+
+## Overview
+
+The *Scheduled Configurations* feature will allow administrators to schedule specific network configurations to be applied and later removed or reconfigured at predefined times without manual intervention. This capability is particularly useful for network scenarios where policies need to change dynamically at certain times of the day or week, such as different ACL rules that should be active only during business hours. The scheduled configuration feature easily integrates within the existing infrastructure of SONiC by following established practices and utilizing widely adopted tools like `cron`. `cron` is a time-based job scheduler in Unix-like operating systems, used to automate the execution of scripts and commands at specified times. This approach not only minimizes the learning curve associated with the adoption of new features but also leverages the proven reliability and efficiency of an open-source utility.
```

Comment:
Just a non-fancy proposal: why not keep this controller outside sonic device, which keeps a gnmi connection and automatically send the gnmi SetRequest at sceduled time. The benefit are:
1. clear seperation of network switch devices and intelligent controller
2. prompty SetResponse to the controoler, and it could follow up with mitigation, auto fix or alert if any failures happen
3. deterministic ConfigDB, mostly likely running config will be compared with golden config, and any diff will be monitored in a large deployer system.